exporter/exporterhelper/internal/persistent_storage.go

```diff
@@ -250,9 +283,16 @@ func (pcs *persistentContiguousStorage) getNextItem(ctx context.Context) (Reques
 		pcs.itemDispatchingStart(ctx, index)
 
 		var req Request
-		batch, err := newBatch(pcs).get(pcs.itemKey(index)).execute(ctx)
+		var bytesSize uint64
+		itemKey := pcs.itemKey(index)
+		batch, err := newBatch(pcs).get(itemKey).execute(ctx)
 		if err == nil {
-			req, err = batch.getRequestResult(pcs.itemKey(index))
+			req, err = batch.getRequestResult(itemKey)
+			if err == nil {
+				bytesSize, err = batch.getSizeByKey(itemKey)
+				pcs.bytesCount.Sub(bytesSize)
+				pcs.updateBytesSize(ctx)
+			}
```

Comment:
There's a problem here. If getting the data fails for whatever reason, then we've already updated our read index and won't ever see this batch again. This didn't matter that much until now, because in case of error we delete it anyway. But now after adding this, our `bytesCount` is going to be larger than it should be forever.

In order to do this correctly, I think we'd need to fold all the updates into a single transaction, which would need to go into a separate PR. So maybe we need some more cleanups for the queue before this functionality can be added. 😒

Reply:
Now that I think about it, I recall having the same doubts.

Reply:
The refactor: https://github.com/aboguszewski-sumo/opentelemetry-collector/commit/43989062725d72cd1d90d50fede2cbf9dc45f38a